packages/zowe-explorer/__tests__/__unit__/dataset/DatasetTree.unit.test.ts

```diff
@@ -340,149 +344,152 @@ describe("Dataset Tree Unit Tests - Function getChildren", () => {
-            collapsibleState: vscode.TreeItemCollapsibleState.None,
-            parentNode: testTree.mSessionNodes[1],
-            profile: blockMocks.imperativeProfile,
-            contextOverride: globals.DS_MIGRATED_FILE_CONTEXT,
-        });
-        const sampleChildren: ZoweDatasetNode[] = [nodeOk, nodeImpError, nodeMigrated];
-        sampleChildren[0].command = { command: "zowe.ds.ZoweNode.openPS", title: "", arguments: [sampleChildren[0]] };
-
-        const children = await testTree.getChildren(testTree.mSessionNodes[1]);
-        expect(children.map((c) => c.label)).toEqual(sampleChildren.map((c) => c.label));
-        expect(children).toEqual(sampleChildren);
-        spyOnDataSetsMatchingPattern.mockRestore();
-    });
-    it("Checking that we fallback to old dataSet API if newer dataSetsMatchingPattern does not exist", async () => {
-        createGlobalMocks();
-        const blockMocks = createBlockMocks();
-
-        const mockMvsApi = await ZoweExplorerApiRegister.getMvsApi(blockMocks.profile);
-        mockMvsApi.dataSetsMatchingPattern = null;
-        const getMvsApiMock = jest.fn();
-        getMvsApiMock.mockReturnValue(mockMvsApi);
-        ZoweExplorerApiRegister.getMvsApi = getMvsApiMock.bind(ZoweExplorerApiRegister);
-
-        const spyOnDataSetsMatchingPattern = jest.spyOn(zosfiles.List, "dataSetsMatchingPattern");
-        const spyOnDataSet = jest.spyOn(zosfiles.List, "dataSet");
-        spyOnDataSet.mockResolvedValueOnce({
-            success: true,
-            commandResponse: null,
-            apiResponse: {
-                items: [{ dsname: "HLQ.USER", dsorg: "PS" }],
-            },
-        });
-        mocked(vscode.window.createTreeView).mockReturnValueOnce(blockMocks.treeView);
-        mocked(Profiles.getInstance).mockReturnValue(blockMocks.profile);
-        blockMocks.datasetSessionNode.pattern = "test";
-        const testTree = new DatasetTree();
-        testTree.mSessionNodes.push(blockMocks.datasetSessionNode);
-        testTree.mSessionNodes[1].dirty = true;
-        const sampleChildren: ZoweDatasetNode[] = [
-            new ZoweDatasetNode({
-                label: "HLQ.USER",
-                collapsibleState: vscode.TreeItemCollapsibleState.None,
-                parentNode: testTree.mSessionNodes[1],
-                profile: blockMocks.imperativeProfile,
-            }),
-        ];
-        sampleChildren[0].command = { command: "zowe.ds.ZoweNode.openPS", title: "", arguments: [sampleChildren[0]] };
-
-        const children = await testTree.getChildren(testTree.mSessionNodes[1]);
-        expect(children.map((c) => c.label)).toEqual(sampleChildren.map((c) => c.label));
-        expect(children).toEqual(sampleChildren);
-        expect(spyOnDataSet).toHaveBeenCalled();
-        expect(spyOnDataSetsMatchingPattern).not.toHaveBeenCalled();
-        spyOnDataSet.mockRestore();
-        spyOnDataSetsMatchingPattern.mockRestore();
-    });
+    // TODO: FIXME
```

Comment:
similar to a previous comment about skipped/commented tests 😋

Reply:
I chose to comment out this test because the test setup had oversights - the test is not ideal because it creates a circular structure for the parent node, which will not occur in normal ZE scenarios. I can try to resolve this, but since the coverage is roughly the same and the test case is unclear anyway ("checking common run of function"), I'll likely remove it unless there are any objections. 🙂

Reply:
I think you've done a pretty good job at keeping up with code-coverage.
Feel free to remove any commented out tests once you are done 😋